knora/dsplib/utils/excel_to_json_properties.py

```diff
@@ -30,75 +33,110 @@ def validate_properties_with_schema(json_file: str) -> bool:
+            attr, val = [x.strip() for x in pair.split(':')]
+            if re.search(r'^\d+\.\d+$', val):
+                val = float(val)
+            elif re.search(r'^\d+$', val):
+                val = int(val)
+            gui_attributes[attr] = val
+
+    # build the dict structure of this property and append it to the list of properties
+    _property = {
+        "name": name,
+        "super": supers,
+        "object": _object,
+        "labels": labels}
+    if comments:
+        _property["comments"] = comments
+    _property["gui_element"] = gui_element
+    if gui_attributes:
+        _property["gui_attributes"] = gui_attributes
+
+    return _property
+
+
+def properties_excel2json(excelfile: str, outfile: str) -> None:
     """
-    Parses the row of an Excel sheet and makes a property from it
+    Converts properties described in an Excel file into a properties section which can be integrated into a DSP ontology
 
     Args:
-        row: the row of an Excel sheet
+        excelfile: path to the Excel file containing the properties
+        outfile: path to the output JSON file containing the properties section for the ontology
 
     Returns:
-        prop (JSON): the property in JSON format
+        None
     """
-    name, super_, object_, en, de, fr, it, comment_en, comment_de, comment_fr, comment_it, gui_element, hlist = row
-    labels = {}
-    if en:
-        labels['en'] = en
-    if de:
-        labels['de'] = de
-    if fr:
-        labels['fr'] = fr
-    if it:
-        labels['it'] = it
-    if not labels:
-        raise Exception(f"No label given in any of the four languages: {name}")
-    comments = {}
-    if comment_en:
-        comments['en'] = comment_en
-    if comment_de:
-        comments['de'] = comment_de
-    if comment_fr:
-        comments['fr'] = comment_fr
-    if comment_it:
-        comments['it'] = comment_it
-    prop = {
-        'name': name,
-        'super': [super_],
-        'object': object_,
-        'labels': labels,
-        'comments': comments,
-        'gui_element': gui_element
-    }
-    if hlist:
-        prop['gui_attributes'] = {'hlist': hlist}
-    return prop
+    
+    # load file
+    df: pd.DataFrame = pd.read_excel(excelfile)
+    df = _prepare_dataframe(
+        df=df,
+        required_columns=["name", "super", "object", "gui_element"],
+        location_of_sheet=f"File '{excelfile}'")
+
+    # transform every row into a property
+    props: list[dict[str, Any]] = list()
+    for i, row in df.iterrows():
+        prop = _row2prop(row, i, excelfile)
+        props.append(prop)
+
+    # write final list to JSON file if list passed validation
+    if _validate_properties_with_schema(json.loads(json.dumps(props, indent=4))):
+        with open(file=outfile, mode='w+', encoding='utf-8') as file:
+            file.write('"properties": ')
+            json.dump(props, file, indent=4)
```

Comment:
is there actually a reason, why this turned into "pseudo-JSON" by adding `"properties": ` in front of it?

Reply:
I think it's to be user friendly. This helps the user to better understand where exactly in the JSON s/he has to add the output. But besides, there is no real reason. It could theoretically be changed.

Reply:
fair enough. I don't think it matters much